MergerLogic/DataTypes/Data.cs

```diff
@@ -55,21 +64,22 @@ public abstract class Data<TUtilsType> : IData where TUtilsType : IDataUtils
         protected TileConvertorFunction ConvertOriginTile;
         protected ValFromCoordFunction ConvertOriginCoord;
 
-        protected Data(IServiceProvider container, DataType type, string path, int batchSize, bool isOneXOne = false, GridOrigin origin = GridOrigin.UPPER_LEFT)
+        protected Data(IServiceProvider container, DataType type, string path, int batchSize, Grid? grid, GridOrigin? origin)
         {
             this.Type = type;
             this.Path = path;
             this.BatchSize = batchSize;
             var utilsFactory = container.GetRequiredService<IUtilsFactory>();
             this.Utils = utilsFactory.GetDataUtils<TUtilsType>(path);
             this.GeoUtils = container.GetRequiredService<IGeoUtils>();
-            this.IsOneXOne = isOneXOne;
-            this.Origin = origin;
+            this.Grid = grid is null ? DefaultGrid() : (Grid)grid.Value;
+            this.IsOneXOne = this.Grid == Grid.OneXOne;
```

Comment:
this will cause problems if the grid is changed after the constructor.
you can either:
- make it run every call (not recommended)
- set it in the grid setter
- make the grid "private init" instead of "protected set" (will prevent modifications after it is initialized)

Reply:
Changed 'IsOneXOne' setter.